Common/Brokerages/ZerodhaBrokerageModel.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data.Market;
+using QuantConnect.Orders;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Orders.TimeInForces;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using static QuantConnect.StringExtensions;
+
+
+namespace QuantConnect.Brokerages
+{
+    public class ZerodhaBrokerageModel : DefaultBrokerageModel
+    {
+        private readonly Type[] _supportedTimeInForces =
+        {
+            typeof(GoodTilCanceledTimeInForce),
+            typeof(DayTimeInForce),
+            typeof(GoodTilDateTimeInForce)
+        };
+
+        private const decimal _maxLeverage = 7m;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ZerodhaBrokerageModel"/> class
+        /// </summary>
+        /// <param name="accountType">The type of account to be modelled, defaults to
+        /// <see cref="AccountType.Margin"/></param>
+        public ZerodhaBrokerageModel(AccountType accountType = AccountType.Margin) : base(accountType)
+        {
+        }
+
+      
+        public override AccountType AccountType => base.AccountType;
+
+        public override decimal RequiredFreeBuyingPowerPercent => base.RequiredFreeBuyingPowerPercent;
+
+        public override void ApplySplit(List<OrderTicket> tickets, Split split)
+        {
+            base.ApplySplit(tickets, split);
+        }
+
+        /// <summary>
+        /// Returns true if the brokerage would be able to execute this order at this time assuming
+        /// market prices are sufficient for the fill to take place. This is used to emulate the
+        /// brokerage fills in backtesting and paper trading. For example some brokerages may not perform
+        /// executions during extended market hours. This is not intended to be checking whether or not
+        /// the exchange is open, that is handled in the Security.Exchange property.
+        /// </summary>
+        /// <param name="security"></param>
+        /// <param name="order">The order to test for execution</param>
+        /// <returns>True if the brokerage would be able to perform the execution, false otherwise</returns>
+        public override bool CanExecuteOrder(Security security, Order order)
+        {
+            return order.SecurityType != SecurityType.Base;
```

Comment:
I think it's better to explicitly list the supported security types here (as done in `CanSubmitOrder`)